docs/syntax_and_semantics/assignment.md

```diff
@@ -11,45 +20,58 @@ local = 1
 
 # Assigns to a class variable
 @@class = 3
+
+# Assigns to a constant
+CONST = 4
+
+# Assigns to a setter method
+foo.method = 5
+foo[0] = 6
 ```
 
-Each of the above kinds of variables will be explained later on.
+### Method as assignment target
 
-Some syntax sugar that contains the `=` character is available:
+A method ending with an equals sign (`=`) is called a setter method. It can be used
+as the target of an assignment. The semantics of the assignment operator apply as
+a form of syntax sugar to the method call.
 
-```crystal
-local += 1  # same as: local = local + 1
+This is an example of a call to a setter method in typical method notation and with assignment operator.
+Both expressions are equivalent.
 
-# The above is valid with these operators:
-# +, -, *, /, %, |, &, ^, **, <<, >>
+```crystal
+def name=(value);end
 
-local ||= 1 # same as: local || (local = 1)
-local &&= 1 # same as: local && (local = 1)
+name=("John")
+name = "John"
 ```
 
-A method invocation that ends with `=` has syntax sugar:
+This is an example of a call to an indexed assignment method in typical method notation and with index assignment operator.
+Both expressions are equivalent.
 
 ```crystal
-# A setter
-person.name=("John")
+class List
+  def []=(key, value); end
+end
+list = List.new
 
-# The above can be written as:
-person.name = "John"
+list.[]=(2, 3)
+list[2] = 3
+```
 
-# An indexed assignment
-objects.[]=(2, 3)
+### Combined assignments
 
-# The above can be written as:
-objects[2] = 3
+[Combined assignments](operators.md#combined-assignments) are a combination of an
+assignment operator and another operator.
+This works with any target type except constants.
 
-# Not assignment-related, but also syntax sugar:
-objects.[](2, 3)
+Some syntax sugar that contains the `=` character is available:
 
-# The above can be written as:
-objects[2, 3]
+```crystal
+local += 1  # same as: local = local + 1
 ```
 
-The `=` operator syntax sugar is also available to setters and indexers. Note that `||` and `&&` use the `[]?` method to check for key presence.
+The `=` operator syntax sugar is also available to setter and index assignment methods.
```

Comment:
Add to this sentence something like "(and this assumes that the corresponding getter methods exist)"